app/recordtransfer/apps.py

```diff
@@ -252,6 +252,12 @@ def verify_caais_defaults():
     if not settings.CAAIS_DEFAULT_CREATION_TYPE:
         raise ImproperlyConfigured("CAAIS_DEFAULT_CREATION_TYPE is not set")
 
+def verify_site_id():
+    """Verify that the SITE_ID setting is valid."""
+    from django.contrib.sites.models import Site  # Import here to avoid circular import
+    site_id = settings.SITE_ID
+    if not Site.objects.filter(pk=site_id).exists():
+        raise ImproperlyConfigured(f"Site with ID {site_id} does not exist.")
```

Comment:
There should be some text here to explicitly mention what setting is causing this issue

```suggestion
        raise ImproperlyConfigured(f"Site with ID {site_id} does not exist. Check the configured SITE_ID")
```